gui/canvaspainter/v7/src/TCanvasPainter.cxx

```diff
@@ -247,12 +233,14 @@ void ROOT::Experimental::TCanvasPainter::CancelCommands(unsigned connid)
 {
    auto iter = fCmds.begin();
    while (iter != fCmds.end()) {
-      auto curr = iter++;
-      if (!connid || (curr->fConnId == connid)) {
-         if (fWaitingCmdId == curr->fId)
-            fWaitingCmdId.clear();
-         curr->CallBack(false);
-         fCmds.erase(curr);
+      auto cmd = *iter;
```

Comment:
If you like such syntax - I can replace it. But `std::list` has simple meaning here - all commands processed sequentially and normally first command always removed with pop_front(). With your syntax I could use `std::queue` instead.

Reply:
`queue` is also that nice - I cannot use simple `for` loop for it like you doing it.
I propose to keep `std::list` with use of iterator - for me it is most efficient solution.

Reply:
- As you know it's "fragile" to remove elements from arbitrary containers while iterating; this only works for some (and yes, list is one of those).
- The pattern you use is generally not more performant than moving elements to a temporary "keep those" container and then swapping the containers' storage.
- I find the resulting algorithm much easier to read, because I don't need to think about "which container type is this, where does the iterator point to after deletion", but you have a clean separation of "read" and "write" container.